homeassistant/components/linky/sensor.py

```diff
@@ -49,45 +52,57 @@
     }
 )
 
+_LOGGER.error("LINKY_SENSOR")
+
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Configure the platform and add the Linky sensor."""
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    timeout = config[CONF_TIMEOUT]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way of setting up the Linky platform."""
+    pass
 
-    account = LinkyAccount(hass, add_entities, username, password, timeout)
-    add_entities(account.sensors, True)
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Add Linky entries."""
+    _LOGGER.error("LINKY_SENSOR:async_setup_entry")
+    account = LinkyAccount(
+        hass,
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        entry.data[CONF_TIMEOUT],
+    )
+
+    account.update_linky_data(dt_util.utcnow())
+
+    sensors = [
+        LinkySensor("Linky yesterday", account, DAILY, INDEX_LAST),
+        LinkySensor("Linky current month", account, MONTHLY, INDEX_CURRENT),
+        LinkySensor("Linky last month", account, MONTHLY, INDEX_LAST),
+        LinkySensor("Linky current year", account, YEARLY, INDEX_CURRENT),
+        LinkySensor("Linky last year", account, YEARLY, INDEX_LAST),
+    ]
+
+    async_track_time_interval(hass, account.update_linky_data, SCAN_INTERVAL)
+
+    async_add_entities(sensors, True)
 
 
 class LinkyAccount:
     """Representation of a Linky account."""
 
-    def __init__(self, hass, add_entities, username, password, timeout):
+    def __init__(self, hass, username, password, timeout):
         """Initialise the Linky account."""
+
+        _LOGGER.error("LINKY_SENSOR:account init")
+        self.hass = hass
         self._username = username
-        self.__password = password
+        self._password = password
         self._timeout = timeout
         self._data = None
-        self.sensors = []
-
-        self.update_linky_data(dt_util.utcnow())
-
-        self.sensors.append(LinkySensor("Linky yesterday", self, DAILY, INDEX_LAST))
-        self.sensors.append(
-            LinkySensor("Linky current month", self, MONTHLY, INDEX_CURRENT)
-        )
-        self.sensors.append(LinkySensor("Linky last month", self, MONTHLY, INDEX_LAST))
-        self.sensors.append(
-            LinkySensor("Linky current year", self, YEARLY, INDEX_CURRENT)
-        )
-        self.sensors.append(LinkySensor("Linky last year", self, YEARLY, INDEX_LAST))
-
-        track_time_interval(hass, self.update_linky_data, SCAN_INTERVAL)
 
     def update_linky_data(self, event_time):
```

Comment:
If we make `event_time` optional and default to `None` we don't need to pass an argument when we call it manually.